include/vast/Translation/CodeGenFallBackVisitor.hpp

```diff
@@ -56,13 +58,16 @@ namespace vast::cg
     template< typename Derived
         , template< typename > typename DefaultMixin
         , template< typename > typename FallBackMixin = DefaultFallBackVisitorMixin
+        , template< typename > typename UnsupporteFallbackdMixin = UnsupportedFallBackVisitor
     >
     struct CodeGenFallBackVisitorMixin
         : DefaultMixin< Derived >
         , FallBackMixin< Derived >
+        , UnsupporteFallbackdMixin< Derived >
```

Comment:
This should be parameter not part of this class, ie.e you pass `UnsupporteFallbackdMixin` as `FallBackMixin`.

I can fix this and integrate to codegen with option if you like?

Reply:
@xlauko, Do you mean creating a config that takes `UnsupportedFallBackVisitor` as one of the parameter. Something like defining the config as:
```
    template< typename Derived >
    using DefaultCodeGenVisitorConfig = CodeGenFallBackVisitorMixin< Derived,
        DefaultCodeGenVisitorMixin,
        UnsupportedFallBackVisitor
    >;
```

One of the reason I didn't add it to `DefaultCodeGenVisitorConfig` because I want both unsupported visitor and default fallback visitor. Previous I updated `DefaultFallBackVisitorMixin` and passed a config as parameter that will multiplex between Unsupported Visitor and Default fallback visitor. It was similar to something like
```
template< typename Derived >
    using DefaultFallbackVisitorConfig = DefaultFallBackVisitorMixin< Derived, UnsupportedFallBackVisitor>;

// pass config to DefaultCodeGenVisitorConfig

    template< typename Derived >
    using DefaultCodeGenVisitorConfig = CodeGenFallBackVisitorMixin< Derived,
        DefaultCodeGenVisitorMixin,
        DefaultFallbackVisitorConfig
    >;
```

This will allow choosing between the default fallback and unsupported visitor in class `DefaultFallBackVisitorMixin` but that mean adding another redirection there.

Can you write a pseudo code of how you like it to be integrated? I will try getting that working else will ask you to update the update it with the options.

Reply:
Thats exactly how it was designed to be used. You don't need to choose anything in `DefaultFallBackVisitorMixin`.
FallBack mixin takes care of trying all visitors in order.

Under the hood it will generate the same code as you have now, which essentially cascades visits `default -> unsupported -> default error`.

Reply:
@xlauko, I added `FallbackVisitorConfig` to chain the unsupported op fallback. I think this is how it is intended to be chained with `DefaultFallBackVisitorMixin`. Let me know if it looks good.